packages/extension-dapp/src/index.ts

```diff
@@ -110,16 +121,18 @@ export async function web3Accounts ({ ss58Format }: Web3AccountsOptions = {}): P
   const injected = await web3EnablePromise;
 
   const retrieved = await Promise.all(
-    injected.map(async ({ accounts, name: source }): Promise<InjectedAccountWithMeta[]> => {
-      try {
-        const list = await accounts.get();
-
-        return mapAccounts(source, list, ss58Format);
-      } catch (error) {
-        // cannot handle this one
-        return [];
+    injected.map(
+      async ({ accounts, name: source }): Promise<InjectedAccountWithMeta[]> => {
+        try {
+          const list = await accounts.get();
+
+          return mapAccounts(source, list.filter((acc) => { return acc.type ? acc.type === accountType : true; }), ss58Format);
```

Comment:
```suggestion
          return mapAccounts(source, list.filter((acc) => acc.type && accountType ? acc.type === accountType : true), ss58Format);
```

If not specified, return all. (NOTE: This keeps it backwards compatible, however... still means that everybody using this needs to adapt)

My only concern now is that it does not allow you to ask for ed25519 & sr25519....

Maybe the cleaner way would be one of -

- excludes as well as include, or
- pass an array of values instead of a single, or
- an explicit isEthereum flag

Thoughts?